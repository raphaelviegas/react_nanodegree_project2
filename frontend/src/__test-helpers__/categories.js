export const initialState = { categories: [] }

export const categories = [
    {
        name: 'react',
        path: 'react'
    },
    {
        name: 'redux',
        path: 'redux'
    },
    {
        name: 'udacity',
        path: 'udacity'
    }
]