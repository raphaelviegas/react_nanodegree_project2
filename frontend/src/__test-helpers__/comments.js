const uuidv4 = require('uuid/v4')

export const initialState = { comments: [] }

export const comments = [
    {
        id: '894tuq4ut84ut8v4t8wun89g',
        parentId: "8xf0y6ziyjabvozdd253nd",
        timestamp: 1468166872634,
        body: 'Hi there! I am a COMMENT.',
        author: 'thingtwo',
        voteScore: 6,
        deleted: false,
        parentDeleted: false
    },
    {
        id: '8tu4bsun805n8un48ve89',
        parentId: "8xf0y6ziyjabvozdd253nd",
        timestamp: 1469479767190,
        body: 'Comments. Are. Cool.',
        author: 'thingone',
        voteScore: -5,
        deleted: false,
        parentDeleted: false
    }
]

export const getCommentsId = '894tuq4ut84ut8v4t8wun89g'

export const removeComment = '894tuq4ut84ut8v4t8wun89g'

export const voteComment = {
    id: '894tuq4ut84ut8v4t8wun89g',
    voteScore: 10
}

let id = uuidv4()
export const addCommentInfo = {
    parentId: '8xf0y6ziyjabvozdd253nd',
    author: 'Test Author',
    body: 'Test Body',
    id
}

export const editCommentInfo = {
    id: '894tuq4ut84ut8v4t8wun89g',
    editContent: {
        body: 'Edited comment body',
        timestamp: Date.now()
    }
}