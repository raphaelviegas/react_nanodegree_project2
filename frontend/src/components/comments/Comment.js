import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Moment from 'react-moment';
import VoteScore from '../shared/VoteScore'

class Comment extends Component {
    static propTypes = {
        commentInfo: PropTypes.object.isRequired,
        onRemoveComment: PropTypes.func.isRequired,
        onEditComment: PropTypes.func.isRequired,
    }

    removeComment(id) {
        this.props.onRemoveComment(id)
    }

    editComment(id, body) {
        this.props.onEditComment(id, body)
    }

    render() {
        const { author, timestamp, body, voteScore, id } = this.props.commentInfo
        return (
            <li>
                <div className="card">
                    <div className="card-content">
                        <span className="card-title comment-title">Author: {author}</span>
                        <span className="card-date comment-date"><Moment format="DD/MMM/YY">{timestamp}</Moment></span>
                        <p className="card-body comment-body">{body}</p>
                    </div>
                    <div className="card-action comment-action">
                        <span className="comment-action-edit" onClick={() => this.editComment(id, body)}>Edit</span>
                        <span className="comment-action-remove" onClick={() => this.removeComment(id)}>Delete</span>
                    </div>
                    <div className="comment-vote-score">
                        <VoteScore size={12} score={voteScore} type='comment' id={id} />
                    </div>
                </div>
            </li>
        )
    }
}

export default Comment
