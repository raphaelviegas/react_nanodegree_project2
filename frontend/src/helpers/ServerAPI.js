
const api = "http://localhost:3001"


// Generate a unique token for storing your bookshelf data on the backend server.

let token = localStorage.token

if (!token)
    token = localStorage.token = Math.random().toString(36).substr(-8)

const headers = {
    'Accept': 'application/json',
    'Authorization': token
}


export const getCategories = () =>
    fetch(`${api}/categories`, { headers })
        .then(res => res.json())
        .then(data => data.categories)

export const getPosts = () =>
    fetch(`${api}/posts`, { headers })
        .then(res => res.json())

// export const getPostsByCategory = (category) =>
//     fetch(`${api}/${category}/posts`, { headers })
//         .then(res => res.json())

// export const getPost = (id) =>
//     fetch(`${api}/posts/${id}`, { headers })
//         .then(res => res.json())

export const getComments = (id) =>
    fetch(`${api}/posts/${id}/comments`, { headers })
        .then(res => res.json())



//General methods for add, edit, vote and remove.
export const add = (path, content) =>
    fetch(`${api}/${path}`, {
        method: 'POST',
        headers: {
            ...headers,
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(content)
    }).then(res => res.json())

export const edit = (path, id, content) =>
    fetch(`${api}/${path}/${id}`, {
        method: 'PUT',
        headers: {
            ...headers,
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(content)
    }).then(res => res.json())

export const vote = (path, id, option) =>
    fetch(`${api}/${path}/${id}`, {
        method: 'POST',
        headers: {
            ...headers,
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(option)
    }).then(res => res.json())

export const remove = (path, id) =>
    fetch(`${api}/${path}/${id}`, {
        method: 'DELETE',
        headers
    }).then(res => res.json())

/* BOOKS API Example

export const get = (bookId) =>
  fetch(`${api}/books/${bookId}`, { headers })
    .then(res => res.json())
    .then(data => data.book)

export const getAll = () =>
  fetch(`${api}/books`, { headers })
    .then(res => res.json())
    .then(data => data.books)

export const update = (book, shelf) =>
  fetch(`${api}/books/${book.id}`, {
    method: 'PUT',
    headers: {
      ...headers,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ shelf })
  }).then(res => res.json())

export const search = (query) =>
  fetch(`${api}/search`, {
    method: 'POST',
    headers: {
      ...headers,
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ query })
  }).then(res => res.json())
    .then(data => data.books)

*/