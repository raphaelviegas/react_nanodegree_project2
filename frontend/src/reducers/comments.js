import {
    LOAD_COMMENTS, 
    REMOVE_COMMENT, 
    VOTE_COMMENT,
    EDIT_COMMENT,
    ADD_COMMENT
} from '../actions/comments'

function commentsReducer (state = {
    comments: [],
}, action) {
    switch (action.type) {
        case LOAD_COMMENTS:
            state = {
                ...state,
                comments: action.payload
            }
            return state
        case REMOVE_COMMENT:
            state ={
                ...state,
                comments: state.comments.filter(comment => comment.id !== action.payload)
            }
            return state
        case VOTE_COMMENT:
            state ={
                ...state,
                comments: state.comments.map(comment => {
                    if (comment.id === action.payload.id) {
                        comment.voteScore = action.payload.voteScore
                    }
                    return comment
                })
            }
            return state
        case ADD_COMMENT:
            let comments = state.comments.concat([action.payload])
            state = {
                ...state,
                comments
            }
            return state
        case EDIT_COMMENT:
            state = {
                ...state,
                comments: state.comments.map(comment => {
                    if (comment.id === action.payload.id) {
                        comment.timestamp = action.payload.editContent.timestamp
                        comment.body = action.payload.editContent.body
                    }
                    return comment
                })
            }
            return state
        default:
            return state
    }
}

export default commentsReducer