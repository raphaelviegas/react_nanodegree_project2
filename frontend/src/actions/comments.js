import * as ServerAPI from '../helpers/ServerAPI'

export const LOAD_COMMENTS = 'LOAD_COMMENTS'
export const REMOVE_COMMENT = 'REMOVE_COMMENT'
export const VOTE_COMMENT = 'VOTE_COMMENT'
export const ADD_COMMENT = 'ADD_COMMENT'
export const EDIT_COMMENT = 'EDIT_COMMENT'

//Local state
export function loadComments (comments) {
  return {
    type: LOAD_COMMENTS,
    payload: comments
  }
}
export function voteComment (id, voteScore) {
  return {
    type: VOTE_COMMENT,
    payload: {id, voteScore }
  }
}

export function removeComment (id) {
  return {
    type: REMOVE_COMMENT,
    payload: id
  }
}

export function addComment(comment) {
  return {
    type: ADD_COMMENT,
    payload: comment
  }
}

export function editComment (id, editContent) {
  return {
    type: EDIT_COMMENT,
    payload: {
      id,
      editContent
    }
  }
}

//Fetch requests
export const fetchComments = (postId) => dispatch => (
  ServerAPI.getComments(postId)
    .then(comments => dispatch(loadComments(comments)))
)

export const fetchVoteComment = (id, option) => dispatch => (
  ServerAPI.vote('comments', id, option)
    .then(comment => dispatch(voteComment(comment.id, comment.voteScore)))
)

export const fetchRemoveComment = (id) => dispatch => (
  ServerAPI.remove('comments', id)
    .then(comment => dispatch(removeComment(comment.id)))
)

export const fetchAddComment = (commentInfo) => dispatch => (
  ServerAPI.add('comments', commentInfo)
    .then(comment => dispatch(addComment(comment)))
)

export const fetchEditComment = (id, editContent) => dispatch => (
  ServerAPI.edit('comments', id, editContent)
    .then(()=> dispatch(editComment(id, editContent)))
)
