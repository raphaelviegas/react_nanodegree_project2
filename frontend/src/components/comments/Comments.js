import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { fetchComments, fetchRemoveComment } from '../../actions/comments'
import MaterialIcon from 'material-icons-react'
import sortBy from 'sort-by'
import AddCommentButton from './addComment/AddCommentButton'
import EditCommentModal from './editComment/EditCommentModal'
import Comment from './Comment'
import M from "materialize-css/dist/js/materialize.min.js"

export class Comments extends Component {
    static propTypes = {
        parentId: PropTypes.string.isRequired,
        comments: PropTypes.array.isRequired
    }

    state = {
        sortByOption: '-voteScore',
        commentId: '',
        commentBody: '',
        editModalOpen: ''
    }

    showEditModal = (id, body) => this.setState(
        () => ({ editModalOpen: 'modal-open', commentId: id, commentBody: body })
    )

    hideEditModal = () => this.setState(
        () => ({ editModalOpen: '', commentId: '', commentBody: '', })
    )

    componentDidMount() {
        this.props.fetchComments(this.props.parentId)
        let elems = document.querySelectorAll('.dropdown-trigger');
        // eslint-disable-next-line
        let instances = M.Dropdown.init(elems, { alignment: 'right', constrainWidth: false, hover: false });
    }

    removeComment = (id) => {
        this.props.fetchRemoveComment(id)
    }

    changeSortCriteria = (option) => {
        switch (option) {
            case 'voteScore':
                this.setState({ sortByOption: 'voteScore' })
                break
            case '-voteScore':
                this.setState({ sortByOption: '-voteScore' })
                break
            case 'timestamp':
                this.setState({ sortByOption: 'timestamp' })
                break
            case '-timestamp':
                this.setState({ sortByOption: '-timestamp' })
                break
            default:
                break
        }
    }

    render() {
        const { comments, parentId } = this.props
        const commentCount = comments.length
        const { sortByOption, commentId, commentBody, editModalOpen } = this.state
        //Sort By
        comments.sort(sortBy(sortByOption))
        return (
            <div className="post-comments">
                <div className="post-comments-title">
                    Comments: {commentCount}
                    <div className="order-by">
                        <a className='dropdown-trigger' data-target='dropdown1'><MaterialIcon icon="sort" size={22} /></a>
                        <ul id='dropdown1' className='dropdown-content'>
                            <li><span>Vote Score</span></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('voteScore')}>Ascending</a></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('-voteScore')}>Descending</a></li>
                            <li className="divider" tabIndex="-1"></li>
                            <li><span>Date</span></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('timestamp')}>Ascending</a></li>
                            <li><a className="order-button" onClick={() => this.changeSortCriteria('-timestamp')}>Descending</a></li>
                        </ul>
                    </div>
                </div>
                <div className="divider"></div>
                {commentCount === 0
                    ? <div className="noComments"><h6>No comments to show</h6></div>
                    : <ul className="comments">
                        {comments.map(comment => (
                            <Comment
                                commentInfo={comment}
                                key={comment.id}
                                onRemoveComment={this.removeComment}
                                onEditComment={this.showEditModal}
                            />
                        ))}
                    </ul>
                }
                <AddCommentButton parentId={parentId} />
                <EditCommentModal
                    modalOpen={editModalOpen}
                    commentId={commentId}
                    commentBody={commentBody}
                    onCloseModal={this.hideEditModal}
                />
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        parentId: ownProps.parentId,
        comments: state.commentsReducer.comments
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchComments: (parentId) => dispatch(fetchComments(parentId)),
        fetchRemoveComment: (id) => dispatch(fetchRemoveComment(id))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Comments));