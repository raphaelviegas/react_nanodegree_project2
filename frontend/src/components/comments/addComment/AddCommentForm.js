import React, { Component } from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'
import { fetchAddComment } from '../../../actions/comments'
const uuidv4 = require('uuid/v4')

export class AddCommentForm extends Component {
    static propTypes = {
        parentId: PropTypes.string.isRequired,
        closeModal: PropTypes.func.isRequired,
    }

    state = {
        body: '',
        author: '',
    }

    onFormChange = (value, inputField) => {
        this.setState({ [inputField]: value })
    }

    onResetForm = () => {
        this.setState({
            body: '',
            author: '',
        })
    }

    onFormSubmit = ({ body, author, parentId }) => {
        let timestamp = Date.now()
        let id = uuidv4()
        let commentInfo = { id, body, author, timestamp, parentId }
        this.props.fetchAddComment(commentInfo)
        this.onResetForm()
        this.props.closeModal()
    }

    render() {
        const { body, author } = this.state
        const { parentId } = this.props
        return (
            <div>
                <div className="modal-content">
                    <h4>Add new comment</h4>
                    <form className="col s12">
                        <div className="row form-row">
                            <div className="input-field col s12 ">
                                <input
                                    placeholder="Author's name"
                                    type="text"
                                    value={author}
                                    onChange={(event) => this.onFormChange(event.target.value, 'author')}
                                />
                                <label>Comment Author</label>
                            </div>
                            <div className="input-field col s12">
                                <textarea
                                    placeholder="Type some text"
                                    type="text"
                                    value={body}
                                    onChange={(event) => this.onFormChange(event.target.value, 'body')}
                                    className="materialize-textarea" />
                                <label>Comment Body</label>
                            </div>
                        </div>
                    </form>
                </div>
                <div className="modal-footer">
                    <button
                        className="red-text text-darken-2 btn-flat addCommentButton btn-reset-comment"
                        onClick={() => this.onResetForm()}
                    >Reset form</button>
                    <button
                        className="btn white-text cyan darken-4 btn-flat addCommentButton btn-add-comment"
                        onClick={() => this.onFormSubmit({ author, body, parentId })}
                    >Add Comment</button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => {
    return {
        parentId: ownProps.parentId,
        closeModal: ownProps.closeModal
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchAddComment: (commentInfo) => dispatch(fetchAddComment(commentInfo))
    }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AddCommentForm));
