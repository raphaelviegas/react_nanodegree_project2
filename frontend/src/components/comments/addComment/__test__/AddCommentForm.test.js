import React from 'react'
import { shallow } from 'enzyme'
import { AddCommentForm } from '../AddCommentForm'
import { addCommentInfo } from '../../../../__test-helpers__/comments'
import { add } from '../../../../helpers/ServerAPI';

describe('<AddCommentForm />', () => {
    const setup = {
        parentId: addCommentInfo.parentId,
        closeModal: jest.fn(),
        fetchAddComment: jest.fn()
    }
    let wrapper
    beforeEach(() => {
        setup.closeModal.mockClear()
        setup.fetchAddComment.mockClear()
        wrapper = shallow(<AddCommentForm {...setup} />)
    })

    it('Should render correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Should have initial state with empty values', () => {
        expect(wrapper.state()).toEqual({ body: '', author: '' })
    })

    it('Should change the state.author after onChange event on author input', () => {
        expect(wrapper.state()).toEqual({ body: '', author: '' })
        wrapper.find('input').simulate('change', { target: { value: 'New Author' } })
        expect(wrapper.state()).toEqual({ body: '', author: 'New Author' })
    })

    it('Should change the state.body after onChange event on textarea', () => {
        expect(wrapper.state()).toEqual({ body: '', author: '' })
        wrapper.find('textarea').simulate('change', { target: { value: 'New Body' } })
        expect(wrapper.state()).toEqual({ body: 'New Body', author: '' })
    })

    it('Should clear the state after clicking on Reset Form Button', () => {
        expect(wrapper.state()).toEqual({ body: '', author: '' })
        wrapper.setState({ body: 'New Body', author: 'New Author' })
        expect(wrapper.state()).toEqual({ body: 'New Body', author: 'New Author' })
        wrapper.find('.btn-reset-comment').simulate('click')
        expect(wrapper.state()).toEqual({ body: '', author: '' })
    })

    it('Should call onFormSubmit when btn-add-comment is clicked', () => {
        const mockonFormSubmit = jest.fn()
        wrapper.instance().onFormSubmit = mockonFormSubmit
        expect(mockonFormSubmit).toHaveBeenCalledTimes(0)
        wrapper.find('.btn-add-comment').simulate('click')
        expect(mockonFormSubmit).toHaveBeenCalledTimes(1)
    })

    it('Should call fetchAddComment, onResetForm and closeModal when method onFormSubmit is called', () => {
        const mockonResetForm = jest.fn()
        wrapper.instance().onResetForm = mockonResetForm
        expect(setup.fetchAddComment).toHaveBeenCalledTimes(0)
        expect(setup.closeModal).toHaveBeenCalledTimes(0)
        expect(mockonResetForm).toHaveBeenCalledTimes(0)
        wrapper.instance().onFormSubmit({ addCommentInfo })
        expect(setup.fetchAddComment).toHaveBeenCalledTimes(1)
        expect(setup.closeModal).toHaveBeenCalledTimes(1)
        expect(mockonResetForm).toHaveBeenCalledTimes(1)
    })
})