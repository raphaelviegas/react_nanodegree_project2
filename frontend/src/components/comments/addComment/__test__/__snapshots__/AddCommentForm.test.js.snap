// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCommentForm /> Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddCommentForm
    closeModal={[Function]}
    fetchAddComment={[Function]}
    parentId="8xf0y6ziyjabvozdd253nd"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal-content"
>
          <h4>
                    Add new comment
          </h4>
          <form
                    className="col s12"
          >
                    <div
                              className="row form-row"
                    >
                              <div
                                        className="input-field col s12 "
                              >
                                        <input
                                                  onChange={[Function]}
                                                  placeholder="Author's name"
                                                  type="text"
                                                  value=""
                                        />
                                        <label>
                                                  Comment Author
                                        </label>
                              </div>
                              <div
                                        className="input-field col s12"
                              >
                                        <textarea
                                                  className="materialize-textarea"
                                                  onChange={[Function]}
                                                  placeholder="Type some text"
                                                  type="text"
                                                  value=""
                                        />
                                        <label>
                                                  Comment Body
                                        </label>
                              </div>
                    </div>
          </form>
</div>,
        <div
          className="modal-footer"
>
          <button
                    className="red-text text-darken-2 btn-flat addCommentButton btn-reset-comment"
                    onClick={[Function]}
          >
                    Reset form
          </button>
          <button
                    className="btn white-text cyan darken-4 btn-flat addCommentButton btn-add-comment"
                    onClick={[Function]}
          >
                    Add Comment
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Add new comment
</h4>,
            <form
              className="col s12"
>
              <div
                            className="row form-row"
              >
                            <div
                                          className="input-field col s12 "
                            >
                                          <input
                                                        onChange={[Function]}
                                                        placeholder="Author's name"
                                                        type="text"
                                                        value=""
                                          />
                                          <label>
                                                        Comment Author
                                          </label>
                            </div>
                            <div
                                          className="input-field col s12"
                            >
                                          <textarea
                                                        className="materialize-textarea"
                                                        onChange={[Function]}
                                                        placeholder="Type some text"
                                                        type="text"
                                                        value=""
                                          />
                                          <label>
                                                        Comment Body
                                          </label>
                            </div>
              </div>
</form>,
          ],
          "className": "modal-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add new comment",
            },
            "ref": null,
            "rendered": "Add new comment",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row form-row"
>
                <div
                                className="input-field col s12 "
                >
                                <input
                                                onChange={[Function]}
                                                placeholder="Author's name"
                                                type="text"
                                                value=""
                                />
                                <label>
                                                Comment Author
                                </label>
                </div>
                <div
                                className="input-field col s12"
                >
                                <textarea
                                                className="materialize-textarea"
                                                onChange={[Function]}
                                                placeholder="Type some text"
                                                type="text"
                                                value=""
                                />
                                <label>
                                                Comment Body
                                </label>
                </div>
</div>,
              "className": "col s12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="input-field col s12 "
>
                    <input
                                        onChange={[Function]}
                                        placeholder="Author's name"
                                        type="text"
                                        value=""
                    />
                    <label>
                                        Comment Author
                    </label>
</div>,
                  <div
                    className="input-field col s12"
>
                    <textarea
                                        className="materialize-textarea"
                                        onChange={[Function]}
                                        placeholder="Type some text"
                                        type="text"
                                        value=""
                    />
                    <label>
                                        Comment Body
                    </label>
</div>,
                ],
                "className": "row form-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <input
                        onChange={[Function]}
                        placeholder="Author's name"
                        type="text"
                        value=""
/>,
                      <label>
                        Comment Author
</label>,
                    ],
                    "className": "input-field col s12 ",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "onChange": [Function],
                        "placeholder": "Author's name",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Comment Author",
                      },
                      "ref": null,
                      "rendered": "Comment Author",
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <textarea
                        className="materialize-textarea"
                        onChange={[Function]}
                        placeholder="Type some text"
                        type="text"
                        value=""
/>,
                      <label>
                        Comment Body
</label>,
                    ],
                    "className": "input-field col s12",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "materialize-textarea",
                        "onChange": [Function],
                        "placeholder": "Type some text",
                        "type": "text",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "textarea",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Comment Body",
                      },
                      "ref": null,
                      "rendered": "Comment Body",
                      "type": "label",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "form",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="red-text text-darken-2 btn-flat addCommentButton btn-reset-comment"
              onClick={[Function]}
>
              Reset form
</button>,
            <button
              className="btn white-text cyan darken-4 btn-flat addCommentButton btn-add-comment"
              onClick={[Function]}
>
              Add Comment
</button>,
          ],
          "className": "modal-footer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset form",
              "className": "red-text text-darken-2 btn-flat addCommentButton btn-reset-comment",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Reset form",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Comment",
              "className": "btn white-text cyan darken-4 btn-flat addCommentButton btn-add-comment",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add Comment",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal-content"
>
            <h4>
                        Add new comment
            </h4>
            <form
                        className="col s12"
            >
                        <div
                                    className="row form-row"
                        >
                                    <div
                                                className="input-field col s12 "
                                    >
                                                <input
                                                            onChange={[Function]}
                                                            placeholder="Author's name"
                                                            type="text"
                                                            value=""
                                                />
                                                <label>
                                                            Comment Author
                                                </label>
                                    </div>
                                    <div
                                                className="input-field col s12"
                                    >
                                                <textarea
                                                            className="materialize-textarea"
                                                            onChange={[Function]}
                                                            placeholder="Type some text"
                                                            type="text"
                                                            value=""
                                                />
                                                <label>
                                                            Comment Body
                                                </label>
                                    </div>
                        </div>
            </form>
</div>,
          <div
            className="modal-footer"
>
            <button
                        className="red-text text-darken-2 btn-flat addCommentButton btn-reset-comment"
                        onClick={[Function]}
            >
                        Reset form
            </button>
            <button
                        className="btn white-text cyan darken-4 btn-flat addCommentButton btn-add-comment"
                        onClick={[Function]}
            >
                        Add Comment
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Add new comment
</h4>,
              <form
                className="col s12"
>
                <div
                                className="row form-row"
                >
                                <div
                                                className="input-field col s12 "
                                >
                                                <input
                                                                onChange={[Function]}
                                                                placeholder="Author's name"
                                                                type="text"
                                                                value=""
                                                />
                                                <label>
                                                                Comment Author
                                                </label>
                                </div>
                                <div
                                                className="input-field col s12"
                                >
                                                <textarea
                                                                className="materialize-textarea"
                                                                onChange={[Function]}
                                                                placeholder="Type some text"
                                                                type="text"
                                                                value=""
                                                />
                                                <label>
                                                                Comment Body
                                                </label>
                                </div>
                </div>
</form>,
            ],
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add new comment",
              },
              "ref": null,
              "rendered": "Add new comment",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="row form-row"
>
                  <div
                                    className="input-field col s12 "
                  >
                                    <input
                                                      onChange={[Function]}
                                                      placeholder="Author's name"
                                                      type="text"
                                                      value=""
                                    />
                                    <label>
                                                      Comment Author
                                    </label>
                  </div>
                  <div
                                    className="input-field col s12"
                  >
                                    <textarea
                                                      className="materialize-textarea"
                                                      onChange={[Function]}
                                                      placeholder="Type some text"
                                                      type="text"
                                                      value=""
                                    />
                                    <label>
                                                      Comment Body
                                    </label>
                  </div>
</div>,
                "className": "col s12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="input-field col s12 "
>
                      <input
                                            onChange={[Function]}
                                            placeholder="Author's name"
                                            type="text"
                                            value=""
                      />
                      <label>
                                            Comment Author
                      </label>
</div>,
                    <div
                      className="input-field col s12"
>
                      <textarea
                                            className="materialize-textarea"
                                            onChange={[Function]}
                                            placeholder="Type some text"
                                            type="text"
                                            value=""
                      />
                      <label>
                                            Comment Body
                      </label>
</div>,
                  ],
                  "className": "row form-row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <input
                          onChange={[Function]}
                          placeholder="Author's name"
                          type="text"
                          value=""
/>,
                        <label>
                          Comment Author
</label>,
                      ],
                      "className": "input-field col s12 ",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "onChange": [Function],
                          "placeholder": "Author's name",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "input",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Comment Author",
                        },
                        "ref": null,
                        "rendered": "Comment Author",
                        "type": "label",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <textarea
                          className="materialize-textarea"
                          onChange={[Function]}
                          placeholder="Type some text"
                          type="text"
                          value=""
/>,
                        <label>
                          Comment Body
</label>,
                      ],
                      "className": "input-field col s12",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "materialize-textarea",
                          "onChange": [Function],
                          "placeholder": "Type some text",
                          "type": "text",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "textarea",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Comment Body",
                        },
                        "ref": null,
                        "rendered": "Comment Body",
                        "type": "label",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "form",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="red-text text-darken-2 btn-flat addCommentButton btn-reset-comment"
                onClick={[Function]}
>
                Reset form
</button>,
              <button
                className="btn white-text cyan darken-4 btn-flat addCommentButton btn-add-comment"
                onClick={[Function]}
>
                Add Comment
</button>,
            ],
            "className": "modal-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset form",
                "className": "red-text text-darken-2 btn-flat addCommentButton btn-reset-comment",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Reset form",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Comment",
                "className": "btn white-text cyan darken-4 btn-flat addCommentButton btn-add-comment",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add Comment",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
